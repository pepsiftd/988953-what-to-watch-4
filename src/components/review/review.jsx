import React from 'react';
import PropTypes from 'prop-types';
import {getRatingString, getFormattedDate} from '@/utils';

const Review = ({review}) => {
  const {user, rating, comment, date} = review;
  const dateString = getFormattedDate(date);

  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{comment}</p>

        <footer className="review__details">
          <cite className="review__author">{user.name}</cite>
          <time className="review__date" dateTime={date.toISOString()}>{dateString}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{getRatingString(rating)}</div>
    </div>
  );
};

Review.propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number,
    user: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date),
  }).isRequired,
};

export {Review};
