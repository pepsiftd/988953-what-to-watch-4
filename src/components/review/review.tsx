import React from 'react';
import {ReviewData} from '@/types';
import {getRatingString, getFormattedDate} from '@/utils';

interface Props {
  review: ReviewData;
}

const Review: React.FunctionComponent<Props> = ({review}) => {
  const {user, rating, comment, date} = review;
  const dateString = getFormattedDate(date);

  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{comment}</p>

        <footer className="review__details">
          <cite className="review__author">{user.name}</cite>
          <time className="review__date" dateTime={date.toISOString()}>{dateString}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{getRatingString(rating)}</div>
    </div>
  );
};

export {Review};
