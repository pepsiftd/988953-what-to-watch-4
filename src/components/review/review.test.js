import React from 'react';
import renderer from 'react-test-renderer';

import {Review} from './review';

const review = {
  id: 1,
  user: {
    id: 4,
    name: `Courtney Cox`,
  },
  rating: 5.1,
  comment: `I don't know what to say but I can't help posting`,
  date: new Date(`August 3, 2020`),
};

it(`Review renders correctly`, () => {
  const tree = renderer.create(
      <Review
        review={review}
      />
  );

  expect(tree).toMatchSnapshot();
});
