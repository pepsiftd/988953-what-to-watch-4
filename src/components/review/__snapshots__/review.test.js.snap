// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      I don't know what to say but I can't help posting
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Courtney Cox
      </cite>
      <time
        className="review__date"
        dateTime="2020-08-02T21:00:00.000Z"
      >
        August 3, 2020
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    5,1
  </div>
</div>
`;
